schemas/data.schema.json

```diff
@@ -57,6 +57,30 @@
           "description": "Short description of the feature, as an HTML string",
           "type": "string"
         },
+        "discouraged": {
+          "additionalProperties": false,
+          "description": "Whether developers are formally discouraged from using this feature",
+          "properties": {
+            "according_to": {
+              "description": "Links to a formal discouragement notice, such as specification text, intent-to-unship, etc.",
+              "items": {
+                "description": "URI",
+                "format": "uri",
+                "type": "string"
+              },
```

Comment:
What do you think about making this an object with both URI and link text?

Having the URI only forces consumers to come up with some link text on their own (or display the full URI, which isn't great).
This would avoid patterns like: "Discouraged, click _here_ to learn why".

Reply:
I think this is a good idea, but I'm slightly concerned about us making guesses about what kind of text that consumers would like. I could imagine several different plausible ways to turn discouragement citations into text (I had a try below). I think it might be better to see how MDN and others actually end up using this and what works with developers, before committing to a single way of doing link text.

I do think we should revisit this, like the reason enums.

---

Possible ways to show this stuff:

- Spec owner:

   > Most developers shouldn't use `with` in new websites (according to [TC39](https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#sec-with-statement)). Consider using [Destructuring](#) instead.
   
- Wikipedia style:

  > Most developers shouldn't use `with` in new websites<sup><a href="https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#sec-with-statement">[1]</a></sup>. Consider using [Destructuring](#) instead.
  

- Fetching the page title:

  > Most developers shouldn't use `with` in new websites (see [ECMAScript® 2025 Language Specification](https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#sec-with-statement)). Consider using [Destructuring](#) instead.

- Domain name:

  > Most developers shouldn't use `with` in new websites ([tc39.es](https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#sec-with-statement)). Consider using [Destructuring](#) instead.

All are possible without a handwritten text field.